core/chains/evm/txm/storage/inmemory_store.go

```diff
@@ -0,0 +1,333 @@
+	m.Lock()
+	defer m.Unlock()
+
+	unconfirmedTx, exists := m.UnconfirmedTransactions[txNonce]
+	if !exists {
+		return fmt.Errorf("unconfirmed tx was not found for nonce: %d - txID: %v", txNonce, txID)
+	}
+
+	// Set the same time for both the tx and its attempt
+	now := time.Now()
+	unconfirmedTx.LastBroadcastAt = now
+	if unconfirmedTx.InitialBroadcastAt.IsZero() {
+		unconfirmedTx.InitialBroadcastAt = now
+	}
+	a, err := unconfirmedTx.FindAttemptByHash(attemptHash)
+	if err != nil {
+		return err
+	}
+	a.BroadcastAt = now
+
+	return nil
+}
+
+func (m *InMemoryStore) UpdateUnstartedTransactionWithNonce(nonce uint64) (*types.Transaction, error) {
+	m.Lock()
+	defer m.Unlock()
+
+	if len(m.UnstartedTransactions) == 0 {
+		m.lggr.Debugf("Unstarted transactions queue is empty for address: %v", m.address)
+		return nil, nil
+	}
+
+	if _, exists := m.UnconfirmedTransactions[nonce]; exists {
+		return nil, fmt.Errorf("an unconfirmed tx with the same nonce already exists: %v", m.UnconfirmedTransactions[nonce])
+	}
+
+	tx := m.UnstartedTransactions[0]
+	tx.Nonce = &nonce
+	tx.State = types.TxUnconfirmed
+
+	m.UnstartedTransactions = m.UnstartedTransactions[1:]
+	m.UnconfirmedTransactions[nonce] = tx
+
+	return tx.DeepCopy(), nil
+}
+
+// Shouldn't call lock because it's being called by a method that already has the lock
+func (m *InMemoryStore) pruneConfirmedTransactions() []uint64 {
+	noncesToPrune := make([]uint64, 0, len(m.ConfirmedTransactions))
+	for nonce := range m.ConfirmedTransactions {
+		noncesToPrune = append(noncesToPrune, nonce)
+	}
+	if len(noncesToPrune) == 0 {
+		return nil
+	}
+	sort.Slice(noncesToPrune, func(i, j int) bool { return noncesToPrune[i] < noncesToPrune[j] })
+	minNonce := noncesToPrune[len(noncesToPrune)/pruneSubset]
+
+	var txIDsToPrune []uint64
+	for nonce, tx := range m.ConfirmedTransactions {
+		if nonce < minNonce {
+			txIDsToPrune = append(txIDsToPrune, tx.ID)
+			delete(m.Transactions, tx.ID)
+			delete(m.ConfirmedTransactions, nonce)
+		}
+	}
+
+	sort.Slice(txIDsToPrune, func(i, j int) bool { return txIDsToPrune[i] < txIDsToPrune[j] })
+	return txIDsToPrune
+}
+
+// Error Handler
+func (m *InMemoryStore) DeleteAttemptForUnconfirmedTx(transactionNonce uint64, attempt *types.Attempt) error {
+	m.Lock()
+	defer m.Unlock()
+
+	tx, exists := m.UnconfirmedTransactions[transactionNonce]
+	if !exists {
+		return fmt.Errorf("unconfirmed tx was not found for nonce: %d - txID: %v", transactionNonce, attempt.TxID)
+	}
+
+	for i, a := range tx.Attempts {
+		if a.Hash == attempt.Hash {
+			tx.Attempts = append(tx.Attempts[:i], tx.Attempts[i+1:]...)
+			return nil
+		}
+	}
+
+	return fmt.Errorf("attempt with hash: %v for txID: %v was not found", attempt.Hash, attempt.TxID)
+}
+
+func (m *InMemoryStore) MarkTxFatal(*types.Transaction) error {
+	return errors.New("not implemented")
+}
+
+// Orchestrator
+func (m *InMemoryStore) FindTxWithIdempotencyKey(idempotencyKey *string) *types.Transaction {
+	m.Lock()
+	defer m.Unlock()
```

Comment:
Think these should be read locks
```suggestion
	m.RLock()
	defer m.RUnlock()
```